compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/DependencyManifestBuilder.java

```diff
@@ -84,7 +84,19 @@ private DependencyManifestBuilder(TomlDocument dependenciesToml,
         this.dependenciesToml = Optional.ofNullable(dependenciesToml);
         this.packageDescriptor = packageDescriptor;
         this.diagnosticList = new ArrayList<>();
-        this.dependencyManifest = parseAsDependencyManifest();
+        DependencyManifest parsedDependecyManifest = parseAsDependencyManifest();
+        if (parsedDependecyManifest.diagnostics().hasErrors()) {
+            var diagnosticInfo = new DiagnosticInfo(
+                    ProjectDiagnosticErrorCode.CORRUPTED_DEPENDENCIES_TOML.diagnosticId(),
+                    "Detected corrupted 'Dependencies.toml' file. This will be updated to latest dependencies.",
```

Comment:
```suggestion
                    "Detected corrupted 'Dependencies.toml' file. This will be updated to the latest dependencies.",
```

also IMO, it's better to say something like,
`Detected corrupted 'Dependencies.toml' file. Dependencies will be updated to the latest versions ignoring the 'Dependencies.toml'.`

Reply:
`Detected corrupted 'Dependencies.toml' file. Dependencies will be updated to the latest versions.`